flair/datasets/document_classification.py

```diff
@@ -1038,26 +923,84 @@ def __init__(
                                         + "\n"
                                     )
 
-        super(NEWSGROUPS, self).__init__(
-            data_folder,
-            tokenizer=tokenizer,
-            memory_mode=memory_mode,
-            **corpusargs,
-        )
+        super(NEWSGROUPS, self).__init__(data_folder, tokenizer=tokenizer, memory_mode=memory_mode, **corpusargs)
 
 
-class SENTIMENT_140(ClassificationCorpus):
+class STACKOVERFLOW(ClassificationCorpus):
     """
-    Twitter sentiment corpus downloaded from and documented at http://help.sentiment140.com/for-students. Two sentiments
-    in train data (POSITIVE, NEGATIVE) and three sentiments in test data (POSITIVE, NEGATIVE, NEUTRAL).
+    Stackoverflow corpus available at "https://github.com/jacoxu/StackOverflow", classifying
+    news items into one of 20 labels. Each data point is a question.
     """
 
     def __init__(
         self,
-        label_name_map=None,
+        base_path: Union[str, Path] = None,
         tokenizer: Tokenizer = SegtokTokenizer(),
         memory_mode: str = "partial",
         **corpusargs,
+    ):
+        """
+        Instantiates Stackoverflow corpus.
+        :param base_path: Provide this only if you store the IMDB corpus in a specific folder, otherwise use default.
+        :param tokenizer: Custom tokenizer to use (default is SegtokTokenizer)
+        :param memory_mode: Set to 'partial' because this is a big corpus, but you can also set to 'full' for faster
+         processing or 'none' for less memory.
+        :param corpusargs: Other args for ClassificationCorpus.
+        """
+
+        if not base_path:
+            base_path = flair.cache_root / "datasets"
+        else:
+            base_path = Path(base_path)
+
+        # this dataset name
+        dataset_name = self.__class__.__name__.lower()
+
+        data_folder = base_path / dataset_name
+
+        # download data if necessary
+        stackoverflow_path_data = (
+            "https://raw.githubusercontent.com/jacoxu/StackOverflow/master/rawText/title_StackOverflow.txt"
+        )
+        stackoverflow_path_label = (
+            "https://raw.githubusercontent.com/jacoxu/StackOverflow/master/rawText/label_StackOverflow.txt"
+        )
+        data_path = flair.cache_root / "datasets" / dataset_name
+        data_file = data_path / "title_StackOverflow.txt"
+        if not data_file.is_file():
+            cached_path(stackoverflow_path_data, Path("datasets") / dataset_name / "original")
+            cached_path(stackoverflow_path_label, Path("datasets") / dataset_name / "original")
+
+            label_list = []
+            # handle labels file
+            with open(data_path / "original" / "label_StackOverflow.txt", "r", encoding="latin1") as open_fp:
+                for line in open_fp:
+                    line = line.rstrip()
+                    label_list.append(line)
```

Comment:
Can you use the textual names of the labels instead of the indices here? This will make it easier to read when printing sentences from the dataset like: 

```python
    corpus = STACKOVERFLOW().downsample(0.1)
    print(corpus.test[0])
```

From the documentation of the dataset, the labels are:

 　   1 wordpress
　　2 oracle
　　3 svn
　　4 apache
　　5 excel
　　6 matlab
　　7 visual-studio
　　8 cocoa
　　9 osx
　　10 bash
　　11 spring
　　12 hibernate
　　13 scala
　　14 sharepoint
　　15 ajax
　　16 qt
　　17 drupal
　　18 linq
　　19 haskell
　　20 magento